File: Answer/3689.最大子数组总值-i.ts
/*
 * @lc app=leetcode.cn id=3689 lang=typescript
 *
 * [3689] 最大子数组总值 I
 */

// @lc code=start
function maxTotalValue(nums: number[], k: number): number {
  let min = nums[0];
  let max = nums[0];
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] < min) {
      min = nums[i];
    }
    if (nums[i] > max) {
      max = nums[i];
    }
  }
  return (max - min) * k;
}
// @lc code=end

File: Answer/3689.最大子数组总值-i.java
/*
 * @lc app=leetcode.cn id=3689 lang=java
 *
 * [3689] 最大子数组总值 I
 */

// @lc code=start
class Solution {

    public long maxTotalValue(int[] nums, int k) {
        long min = nums[0];
        long max = nums[0];
        for (var i = 1; i < nums.length; i++) {
            long v = nums[i];
            if (v < min) {
                min = v;
            }
            if (v > max) {
                max = v;
            }
        }
        return (max - min) * k;
    }
}
// @lc code=end
